File: objc/Visitor/Visitor/Book.m
#import "Book.h"

@implementation Book

-(instancetype)initWith:(NSString*)name noPages:(int)noPages price:(double)price {
    if(self = [super init]) {
        _name = name;
        _noPages = noPages;
        _price = price;
    }
    return self;
}

-(void)accept:(id<LibraryVisitor>)visitor {
    [visitor visitBook:self];
}

@end

File: swift/Visitor/Visitor/Book.swift
import Foundation

class Book {
    var name: String?
    var noPages: Int = 0
    var price: Double = 0
    
    init(_ name: String, noPages: Int, price: Double) {
        self.name = name
        self.noPages = noPages
        self.price = price
    }
}
